OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/AdminControllerTests.cs

```diff
@@ -0,0 +1,780 @@
+    private Mock<ILogger<AdminController>> logger;
+    private Mock<IStringLocalizer<SharedResource>> localizer;
+
+    private string userId;
+    private Guid providerId;
+    private Mock<HttpContext> httpContext;
+
+    private List<ApplicationDto> applications;
+    private IEnumerable<ChildDto> children;
+    private IEnumerable<WorkshopCard> workshops;
+    private ParentDTO parent;
+    private ProviderDto provider;
+    private WorkshopV2Dto workshopDto;
+    private List<MinistryAdminDto> ministryAdminDtos;
+    private List<Provider> providers;
+    private DirectionDto direction;
+    private IMapper mapper;
+    private HttpContext fakeHttpContext;
+
+    [SetUp]
+    public void Setup()
+    {
+        mapper = TestHelper.CreateMapperInstanceOfProfileTypes<CommonProfile, MappingProfile>();
+        currentUserService = new Mock<ICurrentUserService>();
+        sensitiveMinistryAdminService = new Mock<ISensitiveMinistryAdminService>();
+        sensitiveDirectionService = new Mock<ISensitiveDirectionService>();
+        sensitiveProviderService = new Mock<ISensitiveProviderService>();
+        sensitiveApplicationService = new Mock<ISensitiveApplicationService>();
+        logger = new Mock<ILogger<AdminController>>();
+        localizer = new Mock<IStringLocalizer<SharedResource>>();
+
+        userId = Guid.NewGuid().ToString();
+        ministryAdminDtos = AdminGenerator.GenerateMinistryAdminsDtos(10);
+
+        httpContext = new Mock<HttpContext>();
+        httpContext.Setup(c => c.User.FindFirst("sub"))
+            .Returns(new Claim(ClaimTypes.NameIdentifier, userId));
+        
+
+        controller = new AdminController(
+            logger.Object,
+            currentUserService.Object,
+            sensitiveMinistryAdminService.Object,
+            sensitiveApplicationService.Object,
+            sensitiveDirectionService.Object,
+            sensitiveProviderService.Object,
+            localizer.Object)
+        {
+            ControllerContext = new ControllerContext() { HttpContext = httpContext.Object },
+        };
+        providerId = Guid.NewGuid();
+        workshops = FakeWorkshopCards();
+        workshopDto = FakeWorkshop();
+        workshopDto.ProviderId = providerId;
+        children = ChildDtoGenerator.Generate(2).WithSocial(new SocialGroupDto { Id = 1 });
+        providers = ProvidersGenerator.Generate(10);
+        parent = ParentDtoGenerator.Generate().WithUserId(userId);
+        provider = ProviderDtoGenerator.Generate();
+        provider.UserId = userId;
+        provider.Id = providerId;
+        applications = ApplicationDTOsGenerator.Generate(2).WithWorkshopCard(workshops.First()).WithParent(parent);
+        fakeHttpContext = GetFakeHttpContext();
+        controller.ControllerContext.HttpContext = fakeHttpContext;
+    }
+
+    [Test]
+    public async Task GetApplications_WhenCalledByAdmin_ShouldReturnOkResultObject()
```

Comment:
(Test naming) Test states `...WhenCalledByAdmin...` but not sets this prerequisite. Either wrong name or wrong test (or wrong implementation of `GetApplications` method in controller - see my other comment)